src/mlpack/methods/ann/gan.hpp

```diff
@@ -0,0 +1,181 @@
+/**
+ * @file gan.hpp
+ * @author Kris Singh
+ * @author Shikhar Jaiswal
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_GAN_HPP
+#define MLPACK_METHODS_ANN_GAN_HPP
+
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/visitor/output_parameter_visitor.hpp>
+#include <mlpack/methods/ann/visitor/reset_visitor.hpp>
+#include <mlpack/methods/ann/visitor/weight_size_visitor.hpp>
+#include <mlpack/methods/ann/visitor/weight_set_visitor.hpp>
+
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. **/ {
+template<typename Model, typename InitializationRuleType, class Noise>
```

Comment:
Can you add a comment for each template parameter (we can just move up the ones from the constructor), and a simple description what this class does would be great.